docs/source/json-api/index.rst

```diff
@@ -32,127 +32,100 @@ We welcome feedback about the JSON API on `our issue tracker
 
-If the token cannot be read from the provided path or the Ledger API reports an authentication error (for example due to token expiration), the HTTP JSON API will report the error via logging. The token file can be updated with a valid token, and it will be picked up at the next attempt to send a request.
+.. note:: The JSON API provides many other useful configuration flags, run ``daml json-api --help`` to see all of them.
 
-Example session
-***************
+Access Tokens
+=============
 
-.. code-block:: shell
+The JSON API essentially performs two separate tasks:
+
+1. It talks to the IAM of the Ledger API to get data it needs to operate, for this it may need to *provide an access token to an IAM* that requires authentication.
+2. It handles requests coming from one or more Parties, for this each party needs to provide an *access token with each request* it sends to the JSON API.
+
+.. note:: By default sandboxes do not have an IAM and do not need the access token mentioned in point 1
+
+Ledger API Auth with IAM
+------------------------
+
+The IAM access token is used exclusively for maintaining the internal list of known packages and templates.
+
+.. note:: At no point should this access token be provided to an end user, these are for internal use only.
+
+Every IAM access token is different and will depend on what your specific ledger operator's IAM requires.
+The JSON API server requires no access to party-specific data, only access to the ledger identity and package services.
+A token issued for the HTTP JSON API server should contain enough claims to contact these two services but no more than that.
+Please refer to your ledger operator's documentation to find out how.
+
+Once you have retrieved your access token, you can provide it to the JSON API by storing it in a file
+and starting ``daml json-api`` with the flag ``--access-token-file /path/to/your/token.file``.
+
+If the token cannot be read from the provided path or the Ledger API reports an authentication error
+(for example due to token expiration), the JSON API will report the error via logging. 
+
+.. note:: If the token file is updated with a new token it will be picked up at the next attempt to send a request. You can use this to handle cases where an old token expires without restarting your JSON API service.
 
-    $ daml new iou-quickstart-java --template quickstart-java
-    $ cd iou-quickstart-java/
-    $ daml build
-    $ daml sandbox --wall-clock-time --ledgerid MyLedger ./.daml/dist/quickstart-0.0.1.dar
-    $ daml json-api --ledger-host localhost --ledger-port 6865 --http-port 7575
+Party Auth
+----------
 
-Choosing a party
-****************
+Every request from a client to the JSON API requires you to specify a party and some other settings,
+with a JWT token.  HTTP requests pass the token in a header, while WebSocket requests pass the token in a subprotocol.
```

Comment:
```suggestion
Party-specific requests, i.e., command submissions and queries, require a JWT with some additional restrictions compared to the JWT imposed on the format described in <insert link here>: The set of parties listed in `actAs` and `readAs` must contain exactly one party. In addition to that, the application id and ledger id are mandatory. HTTP requests pass the token in a header, while WebSocket requests pass the token in a subprotocol.
```
Not too fussed about the wording but this needs to be documented precisely. It is also not true that all requests need this, e.g., allocating a party just proxies to the underlying service on the ledger API and does not attempt to decode the token.